File: AmazonSDK/AmazonMobileAds/examples/InterstitialAdUniversalSample/InterstitialAdUniversalSample/ViewController.m
// Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// Licensed under the Apache License, Version 2.0 (the "License").
// You may not use this file except in compliance with the License.
// A copy of the License is located at http://aws.amazon.com/apache2.0/
// or in the "license" file accompanying this file.
// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
// CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

#import <AmazonAd/AmazonAdInterstitial.h>
#import <AmazonAd/AmazonAdOptions.h>
#import <AmazonAd/AmazonAdError.h>

#import "ViewController.h"

@interface ViewController () <AmazonAdInterstitialDelegate>

@property (strong, nonatomic) AmazonAdInterstitial *interstitial;
@property (strong, nonatomic) IBOutlet UIButton *loadAdButton;
@property (strong, nonatomic) IBOutlet UIButton *showAdButton;
@property (strong, nonatomic) IBOutlet UILabel *loadStatusLabel;

- (IBAction)loadAmazonInterstitial:(UIButton *)sender;
- (IBAction)showAmazonInterstitial:(UIButton *)sender;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	self.interstitial = [AmazonAdInterstitial amazonAdInterstitial];
    self.interstitial.delegate = self;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)shouldAutorotate
{
    return YES;
}

- (UIInterfaceOrientationMask)supportedInterfaceOrientations
{
    return UIInterfaceOrientationMaskAll;
}

#pragma mark - IBAction

- (IBAction)loadAmazonInterstitial:(UIButton *)sender
{
    self.loadStatusLabel.text = @"Loading interstitial...";
    AmazonAdOptions *options = [AmazonAdOptions options];
    options.isTestRequest = YES;
    [self.interstitial load:options];
}

- (IBAction)showAmazonInterstitial:(UIButton *)sender
{
    [self.interstitial presentFromViewController:self];
}

#pragma mark - AmazonAdInterstitialDelegate

- (void)interstitialDidLoad:(AmazonAdInterstitial *)interstitial
{
    NSLog(@"Interstial loaded.");
    self.loadStatusLabel.text = @"Interstitial loaded.";
}

- (void)interstitialDidFailToLoad:(AmazonAdInterstitial *)interstitial withError:(AmazonAdError *)error
{
    NSLog(@"Interstitial failed to load.");
    self.loadStatusLabel.text = @"Interstitial failed to load.";
}

- (void)interstitialWillPresent:(AmazonAdInterstitial *)interstitial
{
    NSLog(@"Interstitial will be presented.");
}

- (void)interstitialDidPresent:(AmazonAdInterstitial *)interstitial
{
    NSLog(@"Interstitial has been presented.");
}

- (void)interstitialWillDismiss:(AmazonAdInterstitial *)interstitial
{
    NSLog(@"Interstitial will be dismissed.");
}

- (void)interstitialDidDismiss:(AmazonAdInterstitial *)interstitial
{
    NSLog(@"Interstitial has been dismissed.");
    self.loadStatusLabel.text = @"No interstitial loaded.";
}

@end

File: AmazonSDK/AmazonMobileAds/examples/InterstitialAdUniversalSample_Swift/InterstitialAdUniversalSample_Swift/ViewController.swift
// Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// Licensed under the Apache License, Version 2.0 (the "License").
// You may not use this file except in compliance with the License.
// A copy of the License is located at http://aws.amazon.com/apache2.0/
// or in the "license" file accompanying this file.
// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
// CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import UIKit

class ViewController: UIViewController, AmazonAdInterstitialDelegate {

    var interstitial: AmazonAdInterstitial!
    @IBOutlet var loadAdButton: UIButton!
    @IBOutlet var showAdButton: UIButton!
    @IBOutlet var loadStatusLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        interstitial = AmazonAdInterstitial()
        
        interstitial.delegate = self
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func loadAmazonInterstitial(sender: UIButton) {
        loadStatusLabel.text = "Loading Interstitial..."
        
        var options = AmazonAdOptions()
        
        options.isTestRequest = true
        
        interstitial.load(options)
    }

    @IBAction func showAmazonInterstitial(sender: UIButton) {
        interstitial.presentFromViewController(self)    
    }
    
// Mark: - AmazonAdInterstitialDelegate    
    func interstitialDidLoad(interstitial: AmazonAdInterstitial!) {
        Swift.print("Interstitial loaded.")
        loadStatusLabel.text = "Interstitial loaded."
    }
    
    func interstitialDidFailToLoad(interstitial: AmazonAdInterstitial!, withError: AmazonAdError!) {
        Swift.print("Interstitial failed to load.")
        loadStatusLabel.text = "Interstitial failed to load."
    }
    
    func interstitialWillPresent(interstitial: AmazonAdInterstitial!) {
        Swift.print("Interstitial will be presented.")
    }
    
    func interstitialDidPresent(interstitial: AmazonAdInterstitial!) {
        Swift.print("Interstitial has been presented.")
    }

    func interstitialWillDismiss(interstitial: AmazonAdInterstitial!) {
        Swift.print("Interstitial will be dismissed.")
    }
    
    func interstitialDidDismiss(interstitial: AmazonAdInterstitial!) {
        Swift.print("Interstitial has been dismissed.");
        self.loadStatusLabel.text = "No interstitial loaded.";
    }
}

